### PR TITLE
Art generation

### PR DESCRIPTION
**Pull request by salaseviciusm**
- Stress test the generation of art
    - generate 10,000 unique NFTs from a possible 10,368 possible combinations ~ 2s

### CODE REVIEW

File: server/src/controllers/generateArt.ts [hunk 1]
```diff
@@ -39,24 +44,61 @@ function chooseLayerImage(images: ImageI[]): ImageI {
...
+    if (generatedHashes.has(generatedImage.hash)) {
+      // Duplicate made - repeat loop
+      i--;
+      continue;
+    }
+
+    generatedCollection[i] = generatedImage.images;
+    generatedHashes.add(generatedImage.hash);
   }
+  console.log(counter);
```
Comment (Devam-Savjani) [inline server/src/controllers/generateArt.ts:101]: Try avoid console.logging unless necessary?

File: server/src/controllers/generateArt.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ interface GenCollectionI {
...
 }
 
+interface GeneratedImageI {
+  hash: string;
+  images: ImageI[];
+}
+
 function generateRandomPercentage() {
   const MAX_RAND = 100;
   // TODO: make sure this is a uniform distribution with 0-100 inclusive
```
Comment (jaimi-io) [inline server/src/controllers/generateArt.ts:28]: Remove TODO comment

File: server/src/tests/generateArt.test.ts [hunk 3]
```diff
@@ -113,6 +113,53 @@ describe("Generate Art", () => {
...
+    expect(collection[1]).not.toMatchObject(expected);
+  });
+
+  test("Trying to make more NFTs than there are possible combinations of features throws an error", () => {
+    testData.quantity = 100;
+
+    expect(() => generate(testData)).toThrowError();
+  });
+
   // test("TODO", () => {
```
Comment (jaimi-io) [inline server/src/tests/generateArt.test.ts:163]: Remove commented code

File: server/src/controllers/generateArt.ts (hunk 1 above)
Comment (salaseviciusm) [reply server/src/controllers/generateArt.ts:101]: oops yh that was for testing

**Review by alexrichardson1 — APPROVED** [review]

**Review by jaimi-io — APPROVED** [review]

**Pull request merged**